crates/sqlexec/src/remote/rewriter.rs

```diff
@@ -118,15 +153,40 @@ impl TreeNodeRewriter for CopyRemoteToLocalRewriter {
 
     fn mutate(&mut self, node: Self::N) -> Result<Self::N> {
         if let LogicalPlan::Extension(ext) = node {
-            if let Some(copy_to) = ext.node.as_any().downcast_ref::<CopyTo>() {
+            // TODO: Maybe parse extension name here and use match statements?
+            if let Some(insert_into) = ext.node.as_any().downcast_ref::<Insert>() {
+                // Rewrite if the table is temporary.
+                if insert_into.table.meta.is_temp {
+                    self.set(Self::InsertIntoTempTable(insert_into.table.clone()))?;
+                    Ok(insert_into.source.clone())
+                } else {
+                    Ok(LogicalPlan::Extension(ext))
+                }
+            } else if let Some(create_temp_table) =
+                ext.node.as_any().downcast_ref::<CreateTempTable>()
+            {
+                // Rewrite if creating a temporary table.
+                self.set(Self::CreateTempTable(create_temp_table.clone()))?;
+                Ok(create_temp_table
+                    .source
+                    .clone()
+                    .unwrap_or(LogicalPlan::EmptyRelation(EmptyRelation {
+                        produce_one_row: false,
+                        schema: create_temp_table.schema.clone(),
+                    })))
+            } else if let Some(_drop_tables) = ext.node.as_any().downcast_ref::<DropTables>() {
+                // TODO: Figure out how to seperate temp tables and others.
+                // Send the other tables to remote and keep the remote tables
+                // locally. Create a "combined" plan where both remote and local
+                // tables are deleted (if required).
+                Ok(LogicalPlan::Extension(ext))
+            } else if let Some(copy_to) = ext.node.as_any().downcast_ref::<CopyTo>() {
```

Comment:
we should use `ExtensionType` here instead of inline downcasting. It'll make it a bit more readable as well as properly handling all discriminants. 


```rust 
if let LogicalPlan::Extension(ext) = node {
  let ext_type = ext.node.name().parse::<ExtensionType>().unwrap();
  match ext_type {
    ExtensionType::Insert => {
      let insert_into = Insert::try_downcast_extension(ext)?;
      // ..

Reply:
Yeah, there's a TODO at the top for this. I'll make this change as well.